### PR TITLE
[DROOLS-5898] Implementing Serializable on PMML models

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @mariofusco @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-5898

This PR just add "implements Serializable" to AbstractKiePMMLComponent, parent class of all trusty PMML models.
Some modification have been done on (model)ModelImplementationProviderTests to verify it actually works with ClassUtilis.deepClone().

### CODE REVIEW

File: kie-pmml-trusty/kie-pmml-dependencies/pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
   <modelVersion>4.0.0</modelVersion>
 
   <artifactId>kie-pmml-dependencies</artifactId>
-  <packaging>pom</packaging>
+  <packaging>jar</packaging>
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-dependencies/pom.xml]: What is the reason for jar here? I thought this module is just for easily providing of all kie-pmml-dependencies.

**Review by jiripetrlik — COMMENTED** [review]
I have one question. Otherwise looks good.

File: kie-pmml-trusty/kie-pmml-dependencies/pom.xml (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-dependencies/pom.xml]: @jiripetrlik 
As "pom" can't be inserted (via UI) inside Business Central; i.e. BC does not provide a way to declare a dependency different then jar

Comment (jiripetrlik) [reply kie-pmml-trusty/kie-pmml-dependencies/pom.xml]: Understand. Thank you for info. Will approve now.

**Review by jiripetrlik — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]
LGTM :+1:

**Comment by danielezonca** [comment]
@gitgabrio 
Please have a look to this code smell
https://sonarcloud.io/project/issues?id=org.drools%3Adrools&open=AXZsoO_If809r-2lUxD7&pullRequest=3289&resolved=false&types=CODE_SMELL

It seems that `KiePMMLExtension` is not `Serializable`, now it is probably not a problem because we do not support extensions but it makes probably sense to fix it now. Wdyt?

**Comment by gitgabrio** [comment]
> @gitgabrio
> Please have a look to this code smell
> https://sonarcloud.io/project/issues?id=org.drools%3Adrools&open=AXZsoO_If809r-2lUxD7&pullRequest=3289&resolved=false&types=CODE_SMELL
> 
> It seems that `KiePMMLExtension` is not `Serializable`, now it is probably not a problem because we do not support extensions but it makes probably sense to fix it now. Wdyt?

@danielezonca 
Fixed

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins do cdb

**Pull request merged**